docs/notebooks/proton_radiography_particle_tracing.ipynb

```diff
@@ -0,0 +1,227 @@
+{
```

Comment:
<p>I don't get the point of the <code>Grid</code>  scatter plot here. Is it meant just to display the origin of the grid?</p><p> </p><p>Also, the protons could definitely use a <code>label</code>.  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/PlasmaPy/PlasmaPy/pull/895/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
Added a label and changed "Grid" to "Simulated fields" to be more clear.

What I _wanted_ to do was draw a little volume for the simulated fields grid, but that proved complicated and I judged it not worth the space. The point of this graphic is supposed to be just to show that the protons have traveled from the source through the simulated field grid to the detector, and that they all line up nicely in the detector plane at the end.

Reply:
Note to self: look at what fields do the particles see when outside the grid: also, find a civilized way to ask this question.

Reply:
The answer to that question is in this snippet in _push():

`E = self.E[self.xi, self.yi, self.zi, :] * np.outer(self.on_grid, np.ones(3))
        B = self.B[self.xi, self.yi, self.zi, :] * np.outer(self.on_grid, np.ones(3))`

'self.on_grid' is 1 for particles that are on the grid and 0 for particles that are off the grid. So, any particle off the grid sees zero field. 'self.on_grid' itself is calculated right above that: the distance from each particle to its interpolated grid point is calculated and the particle is "on the grid" if it's less than the local grid resolution 'self.ds'.

Reply:
<p>Clever! I like that :)</p>